crates/bevy_pbr/src/meshlet/resource_manager.rs

```diff
@@ -0,0 +1,809 @@
+            }),
+            software_raster_cluster_count: render_device.create_buffer(&BufferDescriptor {
+                label: Some("meshlet_software_raster_cluster_count"),
+                size: size_of::<u32>() as u64,
+                usage: BufferUsages::STORAGE,
+                mapped_at_creation: false,
+            }),
+            raster_cluster_rightmost_slot: cluster_buffer_slots - 1,
+
+            cluster_instance_ids: None,
+            cluster_meshlet_ids: None,
+            second_pass_candidates_buffer: None,
+            depth_pyramid_sampler: render_device.create_sampler(&SamplerDescriptor {
+                label: Some("meshlet_depth_pyramid_sampler"),
+                ..SamplerDescriptor::default()
+            }),
+            depth_pyramid_dummy_texture: render_device
+                .create_texture(&TextureDescriptor {
+                    label: Some("meshlet_depth_pyramid_dummy_texture"),
+                    size: Extent3d {
+                        width: 1,
+                        height: 1,
+                        depth_or_array_layers: 1,
+                    },
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::R32Float,
+                    usage: TextureUsages::STORAGE_BINDING,
+                    view_formats: &[],
+                })
+                .create_view(&TextureViewDescriptor {
+                    label: Some("meshlet_depth_pyramid_dummy_texture_view"),
+                    format: Some(TextureFormat::R32Float),
+                    dimension: Some(TextureViewDimension::D2),
+                    aspect: TextureAspect::All,
+                    base_mip_level: 0,
+                    mip_level_count: Some(1),
+                    base_array_layer: 0,
+                    array_layer_count: Some(1),
+                }),
+
+            previous_depth_pyramids: EntityHashMap::default(),
+
+            // TODO: Buffer min sizes
+            fill_cluster_buffers_bind_group_layout: render_device.create_bind_group_layout(
+                "meshlet_fill_cluster_buffers_bind_group_layout",
+                &BindGroupLayoutEntries::sequential(
+                    ShaderStages::COMPUTE,
+                    (
+                        storage_buffer_read_only_sized(false, None),
+                        storage_buffer_read_only_sized(false, None),
+                        storage_buffer_sized(false, None),
+                        storage_buffer_sized(false, None),
+                    ),
+                ),
+            ),
+            culling_bind_group_layout: render_device.create_bind_group_layout(
+                "meshlet_culling_bind_group_layout",
+                &BindGroupLayoutEntries::sequential(
+                    ShaderStages::COMPUTE,
+                    (
+                        storage_buffer_read_only_sized(false, None),
```

Comment:
That's a lot of bindings, but that does feel a lot easier to read then when we were still using the raw structs 😅

Reply:
Yeahhh I want automated bind groups in the render graph for a reason... Arguably this explicitness is better for perf, but it can't be that much slower to hash-and-cache, and this sucks to write and tweak.